### PR TITLE
hs: update language-javascript

### PR DESCRIPTION
**Pull request by DanBurton**

### CODE REVIEW

**Comment by DanBurton** [comment]
There's currently something wrong with the way I turn `JSConciseBody` into `JSStatement` and/or `[JSStatement]` somewhere. It is causing a TypeMeets_Mismatch on `nim/index-abstract.rsh`, and some sort of impossible situation where it can't find an identifier on `timeoutception/index.txt`.

File: hs/src/Reach/JSUtil.hs [hunk 1]
```diff
@@ -55,6 +58,21 @@ jsa_flatten a = concatMap f a
     f (JSArrayComma _) = []
     f (JSArrayElement e) = [e]
 
+jsConciseBodyToBlock :: JSConciseBody -> JSBlock
+jsConciseBodyToBlock = \case
+  JSConciseFunctionBody b -> b
+  JSConciseExpressionBody e -> JSBlock JSNoAnnot [JSReturn JSNoAnnot (Just e) JSSemiAuto] JSNoAnnot
+
+jsConciseBodyToStmts :: JSConciseBody -> [JSStatement]
+jsConciseBodyToStmts = \case
```
Comment (jeapostrophe) [inline hs/src/Reach/JSUtil.hs:67]: I think I'd make this use `jsConciseBodyToBlock`, then pull them out

File: hs/src/Reach/JSUtil.hs [hunk 2]
```diff
@@ -55,6 +58,21 @@ jsa_flatten a = concatMap f a
...
+  JSConciseFunctionBody (JSBlock _ stmts _) -> stmts
+  JSConciseExpressionBody e -> [JSReturn JSNoAnnot (Just e) JSSemiAuto]
+
+jsConciseBodyToStmt :: JSConciseBody -> JSStatement
+jsConciseBodyToStmt = \case
+  JSConciseFunctionBody (JSBlock l stmts r) -> JSStatementBlock l stmts r JSSemiAuto
+  JSConciseExpressionBody e -> JSStatementBlock JSNoAnnot [JSReturn JSNoAnnot (Just e) JSSemiAuto] JSNoAnnot JSSemiAuto
```
Comment (jeapostrophe) [inline hs/src/Reach/JSUtil.hs:74]: Ditto

**Review by jeapostrophe — COMMENTED** [review]
My guess is that the `JSMethodCall` case isn't behaving the same way with these new `JSConciseBody` things. It is important for it *not* to be a return when the RHS is a method call.

**Pull request merged**